File: backend/src/main/java/com/yourorg/appname/service/CartService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.CartItemRequest;
import com.yourorg.appname.dto.response.CartItemResponse;
import com.yourorg.appname.dto.response.CartSummaryResponse;

public interface CartService {
    CartSummaryResponse getCart(String username);
    CartItemResponse addToCart(CartItemRequest request, String username);
    CartItemResponse updateQuantity(Long cartItemId, Integer quantity, String username);
    void removeFromCart(Long cartItemId, String username);
    void clearCart(String username);
}

File: frontend/src/services/cartService.js
import apiClient from './apiClient';
import { ENDPOINTS } from '../constants/apiEndpoints';

export const cartService = {
  getCart: async () => {
    const response = await apiClient.get(ENDPOINTS.CART);
    return response.data.data;
  },

  addToCart: async (itemData) => {
    const response = await apiClient.post(ENDPOINTS.CART, itemData);
    return response.data.data;
  },

  updateQuantity: async (cartItemId, quantity) => {
    const response = await apiClient.put(`${ENDPOINTS.CART}/${cartItemId}`, null, {
      params: { quantity },
    });
    return response.data.data;
  },

  removeFromCart: async (cartItemId) => {
    const response = await apiClient.delete(`${ENDPOINTS.CART}/${cartItemId}`);
    return response.data;
  },

  clearCart: async () => {
    const response = await apiClient.delete(ENDPOINTS.CART);
    return response.data;
  },
};
